src/spark_manager.py

```diff
@@ -143,16 +151,229 @@ def create_cluster(self,
         # Create or validate postgres config
         postgres_env = self._prepare_postgres_config()
 
-        # TODO: Create the Spark master deployment and service
+        # Create the Spark master deployment and service
+        self._create_master_deployment(master_cores, master_memory, postgres_env)
+        self._create_master_service()
+
         # TODO: Create the Spark worker deployment
 
         # Return cluster information
-        master_url = f"spark://{self.master_name}.{self.namespace}:{self.DEFAULT_MASTER_PORT}"
+        master_url = (
+            f"spark://{self.master_name}.{self.namespace}:{self.DEFAULT_MASTER_PORT}"
+        )
         master_ui_url = f"http://{self.master_name}.{self.namespace}:{self.DEFAULT_MASTER_WEBUI_PORT}"
 
         return {
             "cluster_id": self.cluster_id,
             "master_url": master_url,
             "master_ui_url": master_ui_url,
-            "status": "creating"
+            "status": "creating",
         }
+
+    def _create_master_deployment(
+        self, cores: int, memory: str, postgres_env: List[Dict[str, str]]
+    ):
+        """
+        Create the Spark master deployment.
+
+        Args:
+            cores: Number of CPU cores for the master
+            memory: Memory allocation for the master
+            postgres_env: Postgres environment variables
+        """
+        # Define Spark master environment variables
+        env = [
+            {"name": "SPARK_MODE", "value": "master"},
+            {"name": "SPARK_MASTER_PORT", "value": str(self.DEFAULT_MASTER_PORT)},
+            {
+                "name": "SPARK_MASTER_WEBUI_PORT",
+                "value": str(self.DEFAULT_MASTER_WEBUI_PORT),
+            },
+            {
+                "name": "MAX_EXECUTORS",
+                "value": str(
+                    os.environ.get("MAX_EXECUTORS", self.DEFAULT_MAX_EXECUTORS)
+                ),
+            },
+            {
+                "name": "MAX_CORES_PER_APPLICATION",
+                "value": str(
+                    os.environ.get(
+                        "MAX_CORES_PER_APPLICATION",
+                        self.DEFAULT_MAX_CORES_PER_APPLICATION,
+                    )
+                ),
+            },
+            {
+                "name": "EXECUTOR_CORES",
+                "value": str(
+                    os.environ.get("EXECUTOR_CORES", self.DEFAULT_EXECUTOR_CORES)
+                ),
+            },
+        ] + postgres_env
+
+        # Create the deployment spec
+        # TODO: We could use a YAML file as template for the deployment spec.
+        deployment = {
```

Comment:
I don't really have enough xp with k8s/spark to meaningfully review these data stuctures